pkg/azure/utils.go

```diff
@@ -504,6 +533,15 @@ func (d *MachinePlugin) createVMNicDisk(req *driver.CreateMachineRequest) (*comp
 		vmImageRef = &vmImage
 	}
 
+	if len(providerSpec.Properties.OsProfile.LinuxConfiguration.SSH.PublicKeys.KeyData) == 0 {
+		publicKey, err := generatePublicKey()
+		if err != nil {
+			return nil, err
+		}
+
+		providerSpec.Properties.OsProfile.LinuxConfiguration.SSH.PublicKeys.KeyData = publicKey
```

Comment:
pls print a log of V(3) to inform that a fake ssh key is set to allow VM creation as key not supplied.